File: c/problem_0007.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include "./lib/mathlib.h"

/**
 * @see https://projecteuler.net:
 *
 * 10001st prime
 * Problem 7
 *
 * By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.
 *
 * What is the 10 001st prime number?
 */


int main(void)
{
    int found = 0;
    int stop = 10001;
    unsigned long test = 2;
    while (found < stop) {
        if (isPrime(test)) {
            found++;
        }
        test++;
    }

    printf("the #%i prime is: %lu\n", found, test-1);
}

File: haskell/problem_0007.hs
{--------------------------------------------------------------------
<p>By listing the first six prime numbers: $2, 3, 5, 7, 11$, and $13$, we can see that the $6$th prime is $13$.</p>
<p>What is the $10\,001$st prime number?</p>
---------------------------------------------------------------------}
import EulerLib qualified as E

main =
  let eulerProblem = 7
      solution = last $ take 10001 E.primeList
   in E.printProblemSolution eulerProblem solution
